### PR TITLE
Tighten test suite from audit findings

### PR DESCRIPTION
**Pull request by obenland**
## Summary
- Drop one redundant framework-tester (`test_singleton_returns_same_instance`).
- Rework three existing tests: assert `has_filter()` directly, replace `esc_html__()` with a literal in the conflict-notice assertion, swap a `wp_generate_uuid4()` display name for a static literal.
- Add coverage for three real branches with no current tests: empty-`display_name` skip in activation, page-only scope of `get_page_by_path()`, and the stale-entry guard in `admin_notices()`.

Net change: 14 → 16 tests, no behavior changes in plugin code.

## Test plan
- [ ] CI `phpunit.yml` (PHP 7.4 + 8.4, single-site **and** multisite) passes.
- [ ] CI `wpcs.yml` produces no new annotations.
- [ ] Locally: `npm run start && npm run test-php && npm run test-php-multisite` all green.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: tests/test-wp-author-slug.php [hunk 1]
```diff
@@ -307,12 +304,94 @@ public function test_deactivation_clears_conflicts() {
...
+		$this->assertFalse(
+			(bool) get_option( 'wp_author_slug_conflicts' ),
+			'Expected no conflicts when only a non-page post shares the proposed slug.'
+		);
+	}
+
+	/**
+	 * Tests that `admin_notices` skips conflict entries whose user or page no
+	 * longer exists, rather than emitting a malformed list item or fataling on
+	 * a null user/page.
```
[the remark this answers is not in the document]
Comment (obenland) [reply tests/test-wp-author-slug.php]: Fixed in b400776 — reworded to "triggering a fatal error".
